### ISSUE TITLE
Clarification on fixed τ_f vs Algorithm 1 sampling, and reproducibility of Fig. 8(c) joint-vs-decoupled t-SNE

### ISSUE THREAD

**Issue by xiao10ma** [Mondo-Robotics/DiT4DiT#21]
Hi authors, thanks for releasing the paper and code. I have two clarification / reproducibility questions after reading the paper and the current implementation.

## 1. Fixed feature timestep `τ_f` vs Algorithm 1 line 11

In the paper, the feature extraction timestep `τ_f` is described as fixed, so that the Action DiT receives deterministic / consistent hidden representations from a stable operating point of the Video DiT.

However, Algorithm 1 line 11 appears to sample it every iteration:

```text
τ_f ∼ U{0/T, 1/T, ..., T/T}
```

This seems inconsistent with the “fixed `τ_f`” description. If `τ_f` is resampled per batch / iteration, then the action model would receive hidden states from different video denoising stages, which would no longer be a fixed feature-extraction operating point.

In the released code, I do not see random sampling of `τ_f`. Instead, the implementation seems to always capture hidden states from the first Cosmos denoising step:

```python
def forward(..., num_inference_steps: int = 1, capture_step_index: int = 0, ...)
```

and inside the denoising loop:

```python
sigma_t = self.scheduler.sigmas[i]
...
if i == 0 and hidden_first is None:
    hidden_first = self._cached_hidden[-1]
    self._capture_hidden_enabled = False
```

The interface also does not appear to pass `capture_step_index`, so the default `capture_step_index=0` is used. Therefore the effective feature timestep seems to be:

```text
τ_f = scheduler.sigmas[0]
```

rather than a uniformly sampled timestep.

Could you clarify the intended behavior?

1. Is Algorithm 1 line 11 a typo, and should `τ_f` instead be selected once and kept fixed throughout training and inference?
2. What exact `τ_f` was used in the experiments reported in the paper? Was it always `capture_step_index=0`, i.e. the first scheduler sigma?
3. Did any experiment actually sample `τ_f ∼ U{0/T, ..., T/T}` during training? If yes, how should that be reconciled with the claim that fixed `τ_f` provides stable hidden states for the Action DiT?
4. It would be helpful if the paper / README could explicitly state the actual `τ_f` used in the released implementation.

## 2. Fig. 8(c): joint-vs-decoupled t-SNE and whether action loss updates the Video DiT

I also have a question about the interpretation of Fig. 8(c). The paper describes Fig. 8(c) as a t-SNE visualization of extracted hidden features, color-coded by temporal phase within an episode: Early, Middle, Late. The text says that joint training improves temporal separation and “forces the visual backbone to embed a continuous, physics-aware temporal progression.”

However, in the released code, the hidden features extracted from the Video DiT appear to be detached from the computation graph. In the hook:

```python
self._cached_hidden.append(out.detach())
```

and the extractor is also called with:

```python
detach=True
```

Therefore, as far as I understand the current implementation, the action loss cannot backpropagate through the extracted hidden states into the Video DiT. The training objective may be written as:

```text
total_loss = action_loss + λ * future_video_loss
```

but the gradient paths appear to be effectively:

```text
future_video_loss  -> updates Video DiT
action_loss        -> updates Action DiT
```

rather than:

```text
action_loss -> Action DiT -> hidden states -> Video DiT
```

This makes the interpretation of Fig. 8(c) somewhat unclear.

If Fig. 8(c) visualizes Video DiT hidden states, and if the released implementation with detached hidden states was used, then the action objective does not directly force the visual backbone to organize its features by temporal phase. Any difference between “joint” and “decoupled” representations would instead seem to come from other factors, such as continued online training of the Video DiT with the video-generation objective during action training, different checkpoint timing, different training schedule, or a non-released implementation without the detach.

Could you clarify the exact setup used for Fig. 8(c)?

1. Are the t-SNE points in Fig. 8(c) extracted from the Video DiT hidden states, the Action DiT internal hidden states, or some other action-conditioned representation?
2. Was Fig. 8(c) generated using the released implementation where the extracted Video DiT hidden states are detached, or using a version where action-loss gradients are allowed to flow into the Video DiT?
3. In the “joint” setting for Fig. 8(c), does `action_loss` update the Video DiT, or is the Video DiT updated only by the video-generation / future-video loss?
4. What is the exact “decoupled” protocol used for Fig. 8(c)? Is it:

   * pretrain Video DiT with video generation,
   * freeze Video DiT,
   * then train Action DiT;

   while “joint” means simultaneously training Video DiT with video loss and Action DiT with action loss?
5. Were the decoupled and joint representations compared at matched Video-DiT training budgets? For example, if the joint model continues updating the Video DiT during action training, does it receive more video-loss updates than the decoupled model?
6. For the reported silhouette scores, were the cluster labels Early / Middle / Late? Were the scores computed in the original hidden-feature space or after t-SNE projection?
7. Could you release the script / config used to generate Fig. 8(c), including the checkpoint names, layer index, `τ_f`, number of sampled points, task sampling procedure, and random seed?

I may be misunderstanding the implementation, but this distinction seems important. If action gradients are detached from the Video DiT, then the result is still a form of simultaneous training, but it is different from a fully end-to-end joint optimization where the action objective directly shapes the visual backbone. In that case, the interpretation of Fig. 8(c) would be better described as “continued video-objective training during action training improves temporal structure,” rather than “the action policy objective directly forces the visual backbone to learn temporally organized representations.”

Thanks again for the work and for releasing the code. Clarifying these two points would make the training protocol and representation analysis much easier to reproduce and interpret.

**Comment by ldx825** [comment]
I have the same questions and concerns, especially regarding the fixed versus sampled $τ_f$ , the detached hidden states, and the exact protocol used for Fig. 8(c). These details seem important for correctly understanding whether the reported effect comes from true end-to-end joint optimization or from simultaneously training the two branches with separate gradient paths. I would also greatly appreciate clarification from the authors, as well as the corresponding configuration or script used to reproduce Fig. 8(c). Thank you!
